replica/database.hh

```diff
@@ -495,6 +495,7 @@ private:
 
     class table_state;
     std::unique_ptr<table_state> _table_state;
+    condition_variable _sstables_changed;
```

Comment:
How does regular compaction update _sstables_changed? IIUC the flow is compaction -> compaction_manager -> compaction_strategy. refresh_compound_sstables_set() is not called AFAICT.

Reply:
That's called on compaction completion, when we're done updating the main set. we're watching the sstable set for changes, so this sounds like the right place for it. when compaction starts, there's no change, so waking up the loop will result in an useless check.